tests/src/JIT/Performance/CodeQuality/HWIntrinsic/X86/PacketTracer/VectorPacket.cs

```diff
@@ -10,7 +10,7 @@
 using System.Runtime.CompilerServices;
 using System;
 
-internal class VectorPacket256
+internal struct VectorPacket256
```

Comment:
If you are using `in`, this should probably be `readonly struct` to ensure you don't incur any hidden copies.

Reply:
We need to update `VectorPacket256` fields in the loop https://github.com/dotnet/coreclr/pull/19662/files#diff-68935896f95ba1d13b35f248c35f1e63R173

Reply:
Hmm, seems we can optimize here, will try.

Reply:
Then you may be incurring hidden copies when passing the value as `in`. I would recommend checking this with the `ErrorProne.NET` analyzer

Reply:
> Then you may be incurring hidden copies when passing the value as in.

Probably not, that loop just updates a local struct. Will try to eliminate the field update.

Reply:
I tried to eliminate the field update of local structs via updating local vectors then `new` the VectorPacket by the local vectors. 
This change makes a little bit slower (0.75s vs 0.74s) and a bit code size regression due to callee-saved SIMD register prolog.

So, I think mutable `struct` is okay for this program.

Reply:
BTW, I have updated this PR to use pass-by-value.